streaming_form_data/_parser.pyx

```diff
@@ -152,80 +170,67 @@ cdef class _Parser:
             return 0
 
         cdef bytes chunk
-        cdef long index, buffer_start, buffer_end
+        cdef Index index
 
         if self._leftover_buffer:
             chunk = self._leftover_buffer + data
-
             index = len(self._leftover_buffer)
-            buffer_start = 0
-            buffer_end = index
-
             self._leftover_buffer = None
         else:
             chunk = data
-
             index = 0
-            buffer_start = 0
-            buffer_end = 0
 
-        return self._parse(chunk, index, buffer_start, buffer_end)
+        return self._parse(chunk, index)
 
-    cdef int _parse(self, bytes chunk, long index,
-                    long buffer_start, long buffer_end):
-        cdef long idx, byte
+    cdef int _parse(self, bytes chunk, Index index):
+        cdef Index idx, buffer_start, chunk_len
+        cdef Index match_start, skip_count, matched_length
+        cdef Byte byte
+        cdef const Byte *chunk_ptr
+        chunk_ptr = chunk
+        chunk_len = len(chunk)
+        buffer_start = 0
 
-        for idx in range(index, len(chunk)):
-            byte = chunk[idx]
+        idx = index
+        while idx < chunk_len:
+            byte = chunk_ptr[idx]
 
             if self.state == ParserState.PS_START:
                 if byte != Constants.Hyphen:
-                    return 1
+                    return 10
```

Comment:
Could you document these return values somewhere?

Reply:
I need to think about it. Once we document them, I would like to group error values by their meaning.

Reply:
The number of error codes would roughly be the same as the number of Parser states, right? The grouping could be done on that basis?

Reply:
not exactly. We can have few returns from the same parser state due to different failed assumptions.

Reply:
Sure, but the grouping could be done on that basis anyway. For instance, returns from `PS_START` could be 1x, `PS_STARTING_BOUNDARY` could be 2x, and so on. This way at least we have an idea of what the state the parser was in when things went wrong.